### PR TITLE
chore: update CODEOWNERS for new crate structure

### PR DESCRIPTION
**Pull request by Rjected**
This updates CODEOWNERS for the current folder structure we have, removed some things that no longer exist, and tried to assign people to paths

If anyone wants on / off one of these folders, please comment!

I'm not sure who to put on `fs-util`, so I would like feedback on that

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -1,27 +1,37 @@
...
+crates/primitives/          @DaniPopes @Rjected
+crates/prune/               @shekhirin @joshieDo
 crates/revm/                @mattsse @rakita
 crates/rpc/                 @mattsse @Rjected
 crates/stages/              @onbjerg @rkrasiuk @shekhirin
-crates/stages-api/          @onbjerg @rkrasiuk @shekhirin
-crates/static-file          @joshieDo @shekhirin
+crates/static-file/         @joshieDo @shekhirin
+crates/static-file-types/   @joshieDo @shekhirin
 crates/storage/             @rakita @joshieDo @shekhirin
```
Comment (onbjerg) [inline CODEOWNERS:31]: think we need to unpack this folder at some point, it contains so much code by diff authors i think it makes sense to be granular

Comment (Rjected) [reply CODEOWNERS:31]: any suggestions on owners for each subfolder?

Comment (onbjerg) [reply CODEOWNERS:31]: the owner set will be roughly the same, but i think it makes sense to be granular still, esp when/if these are refactored, so i think

```codeowners
crates/storage/codecs              @joshieDo
crates/storage/db                      @joshieDo @rakita 
crates/storage/db-api               @joshieDo @rakita # think these were written primarily by joshie/dragan?
crates/storage/libmdbx-rs        @rakita # think dragan has primarily been maintaining this
crates/storage/nippy-jar           @joshieDo @shekhirin 
crates/storage/provider            @rakita @joshieDo @shekhirin 
```

storage-api/errors/db-common i have no idea

Comment (onbjerg) [reply CODEOWNERS:31]: no idea why formatting is so bad, they are aligned when editing the comment

**Review by mattsse — APPROVED** [review]

**Review by rkrasiuk — APPROVED** [review]

File: CODEOWNERS (hunk 1 above)
Comment (Rjected) [reply CODEOWNERS]: copied this, tried to fill all the subfolders for storage

**Pull request merged**